library/queue.js

```diff
@@ -0,0 +1,109 @@
+'use strict';
+
+class Queue {
+    //  creating a queue of length = lengthOfQueue
+    // creating a new array of the length provided
+    // setting up head and tail values. Head value will determine location of pop and tail value will determine location of push. 
+    // countOfItems keeps a count of number of items in the array at a given time.
+    constructor(lengthOfQueue) {
+        this.lengthOfQueue = this.verifyLengthInput(lengthOfQueue);
+        this.head = 0;
+        this.tail = 0;
+        this.countOfItems = 0;
+        this.queue = new Array(this.lengthOfQueue);
+    }
+
+    verifyLengthInput(lengthOfQueue) {
+        /** 
+        * Verify the lengthOfQueue provided is an integer.
+        * @param {integer} lengthOfQueue
+        * @return {integer}
+        */
+        if (!(Number.isInteger(lengthOfQueue))) {
+            throw new Error("Only accept integer as length");
+        }
+        return lengthOfQueue;
+    }
+
+    push(value) {
+        /** 
+        * Pushing value onto the queue
+        * If queue is full and head===tail, an error is thrown as no more additions to the queue is allowed. 
+        * The tail value is reset to zero once it equals lengthOfQueue.
+        * @param {any} value
+        * @return {array}
+        */
+        if (this.tail === this.head && this.countOfItems === this.lengthOfQueue) {
```

Comment:
Removed from push, but still required for pop.

Reply:
why is it required ?

Reply:
This is required for both actually, otherwise the functions would fail right from the start. Initially, head and tail are both equal to 0, and if everytime I try to push or pop, it would throw an error (as if cond states if head=tail, throw error).

Reply:
shouldnt you only check for count and see if full or empty ?